frontend/src/components/c-authorship-file.vue

```diff
@@ -0,0 +1,436 @@
+import brokenLinkDisabler from '../mixin/brokenLinkMixin';
+import tooltipPositioner from '../mixin/dynamicTooltipMixin';
+import cSegmentCollection from './c-segment-collection.vue';
+import { AuthorshipFile } from '../types/types';
+import { FilesSortType } from '../types/authorship';
+
+export default defineComponent({
+  name: 'c-authorship-file',
+  components: {
+    cSegmentCollection,
+  },
+  mixins: [brokenLinkDisabler, tooltipPositioner],
+  props: {
+    file: {
+      type: Object as PropType<AuthorshipFile>,
+      required: true,
+    },
+    index: {
+      type: Number,
+      required: true,
+    },
+    filesSortType: {
+      type: String,
+      required: true,
+    },
+    info: {
+      type: Object as PropType<{ repo: string, isMergeGroup: Boolean }>,
+      required: true,
+    },
+    authorColors: {
+      type: Object as PropType<{ [key: string]: string }>,
+      required: true,
+    },
+    fileTypeColors: {
+      type: Object as PropType<{ [key: string]: string }>,
+      required: true,
+    },
+  },
+  emits: ['toggle-file-active-property'],
+  methods: {
+    getHistoryLink(file: AuthorshipFile): string | undefined {
+      const repo = window.REPOS[this.$props.info.repo];
+      return window.getHistoryLink(this.$props.info.repo, repo.branch, file.path);
+    },
+
+    getBlameLink(file: AuthorshipFile): string | undefined {
+      const repo = window.REPOS[this.$props.info.repo];
+      return window.getBlameLink(this.$props.info.repo, repo.branch, file.path);
+    },
+
+    getAuthors(file: AuthorshipFile): (string | null)[] {
+      return Array.from(new Set(file.segments?.map((segment) => segment.knownAuthor)
+        .filter(Boolean))).sort().slice(0, 50);
+    },
+
+    onTitleTooltipHover(tooltipTextElement: string, titleTextElement: string): void {
+      this.onTooltipHover(tooltipTextElement);
+      const titleElement = this.getElementByRef(titleTextElement);
+      titleElement.classList.add('max-zIndex');
+    },
+
+    resetTitleTooltip(tooltipTextElement: string, titleTextElement: string): void {
+      this.resetTooltip(tooltipTextElement);
+      const titleElement = this.getElementByRef(titleTextElement);
+      titleElement.classList.remove('max-zIndex');
+    },
+
+    toggleFileActiveProperty(file: AuthorshipFile): void {
+      this.scrollFileIntoView(file);
+      this.$emit('toggle-file-active-property', file);
+    },
+
+    scrollFileIntoView(file: AuthorshipFile): void {
+      const fileElement = this.getElementByRef(file.path);
+      if (this.isElementAboveViewport(fileElement)) {
+        fileElement.scrollIntoView(true);
+      }
+    },
+
+    isUnknownAuthor(name: string): boolean {
+      return name === '-';
+    },
+
+    addBlankLineCount(fileType: string, lineCount: number, filesInfoObj: { [key: string]: number }): void {
+      if (!filesInfoObj[fileType]) {
+        filesInfoObj[fileType] = 0;
+      }
+
+      filesInfoObj[fileType] += lineCount;
+    },
```

Comment:
Are these two methods used anywhere in the file?

Reply:
Nope, they're mistakenly copied from the parent component. Thanks for spotting that!